tests/test_cli_compile.py

```diff
@@ -2457,6 +2447,38 @@ def test_triple_equal_pinned_dependency_is_used(
 )
 
 
+@pytest.mark.parametrize(("fname", "content"), METADATA_TEST_CASES)
+def test_not_specified_input_file(
+    runner, make_module, fname, content, tmpdir, monkeypatch
+):
+    """
+    It should raise an error if there are no input files or default input files
+    such as "setup.py", "requirements.in" or "pyproject.toml".
+    """
+    make_module(fname=fname, content=content)
+
+    monkeypatch.chdir(tmpdir)
+    out = runner.invoke(cli)
+    monkeypatch.undo()
```

Comment:
This is supposed to happen automatically.
```suggestion
```

Reply:
And it does, but in this case it complains that the path does not exist. I'm guessing that is because the `tmpdir` is automatically removed before that happens.

Reply:
Right..
This is because `monkeypatch` and `tmp_path` don't depend on each other and pytest doesn't guarantee the cleanup sequence in this case.

Reply:
@berislavlopac so a cleaner solution could be to use a context manager (`with monkeypatch.context() as mp_ctx:`). With that, you won't need the explicit `undo()` and there would be a clear block where chdir is applied.

Reply:
Good point, thanks!